src/passes/string-encoding/StringEncoding.cpp

```diff
@@ -494,18 +534,17 @@ bool StringEncoding::processReplace(BasicBlock &BB, Instruction &I, Use &Op,
     New = New.substr(0, Data.getNumElements() - 1);
   }
 
-  Constant *NewVal = ConstantDataArray::getString(BB.getContext(), New);
+  Constant *NewVal = ConstantDataArray::getString(I.getContext(), New);
```

Comment:
The review would be easier to comprehend, if unrelated refactors like these were split off into a separate PR . (Or committed right away as NFC?)

Reply:
Moved the refactor in a separate commit.